coap/client.go

```diff
@@ -53,33 +54,28 @@ func NewClient(c mux.Client, tkn message.Token, l *slog.Logger) Client {
 }
 
 func (c *client) Done() <-chan struct{} {
-	return c.client.Done()
+	return c.client.Conn().Done()
 }
 
 func (c *client) Cancel() error {
-	m := message.Message{
-		Code:    codes.Content,
-		Token:   c.token,
-		Context: context.Background(),
-		Options: make(message.Options, 0, 16),
-	}
-	if err := c.client.WriteMessage(&m); err != nil {
+	pm := c.client.Conn().AcquireMessage(context.Background())
+	pm.SetCode(codes.Content)
+	pm.SetToken(c.token)
+	if err := c.client.Conn().WriteMessage(pm); err != nil {
 		c.logger.Error(fmt.Sprintf("Error sending message: %s.", err))
 	}
-	return c.client.Close()
+	return c.client.Conn().Close()
 }
 
 func (c *client) Token() string {
 	return c.token.String()
 }
 
 func (c *client) Handle(msg *messaging.Message) error {
-	m := message.Message{
-		Code:    codes.Content,
-		Token:   c.token,
-		Context: c.client.Context(),
-		Body:    bytes.NewReader(msg.GetPayload()),
-	}
+	pm := pool.NewMessage(context.Background())
```

Comment:
Why not use client connection context
```suggestion
	pm := pool.NewMessage(c.client.Conn().Context())
```